File: 35-search-insert-position/35-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        nums.push_back(target);
        sort(nums.begin()  , nums.end());
        vector<int>::iterator it ;
        it = find(nums.begin() , nums.end() , target);
        int x = it-nums.begin();
        return x;
    }
};

File: 35-search-insert-position/35-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        
        int start = 0;
        int end = nums.length-1;
        int mid = start + (end-start)/2;
        
        if(Arrays.binarySearch(nums , target)>=0){
            return Arrays.binarySearch(nums , target);
        }

        else{
            while(start<=end){
                if(nums[mid]<target){
                    start=mid+1;
                }
                else{
                    end=mid-1;
                }
                 mid = start + (end-start)/2;
            }
        }
        return mid;
    }
}